lib/src/main/java/org/team100/lib/sensors/Team100CANIMU.java

```diff
@@ -0,0 +1,88 @@
+package frc.robot;
+
+import edu.wpi.first.hal.CANData;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.wpilibj.CAN;
+
+public class Team100CANIMU {
+    public static final int CAN_DEVICE_MANUFACTURER = 8;
+    public static final int CAN_DEVICE_TYPE = 4;
+    public static final int API_ID_EULER_ANGLES = 17;
+    public static final int API_ID_QUATERNION = 18;
+    // the 16 bit values on the CAN bus must be divided by 163.84 then 200 should be
+    // subtracted from the value
+    public static final double INT_TO_ENG_COEFF = 1 / 163.84;
+    public static final double INT_TO_ENG_OFFSET = -200;
+    public static final double INT_TO_ENG_ACC = 1 / 655.36;
+
+    private long rollValueInt, pitchValueInt, yawValueInt, accValueInt;
+    private int deviceID;
+    private boolean result;
+    private CAN Team100CANIMU;
+    private CANData IMUData;
+    private EulerAngles EAngles;
+
+    /*
+     * Constructor
+     */
+    public Team100CANIMU(int deviceID) {
+        this.deviceID = deviceID;
+        Team100CANIMU = new CAN(this.deviceID, CAN_DEVICE_MANUFACTURER, CAN_DEVICE_TYPE);
+        IMUData = new CANData();
+        EAngles = new EulerAngles();
+
+    }
+
+    /*
+     * Destructor
+     */
+//  public void close() {
+//    Team100CANIMU.close();
+//  }
+
+    /*
+     * Get Euler Angles
+     */
+    public EulerAngles getEulerAngles() {
+        result = Team100CANIMU.readPacketLatest(API_ID_EULER_ANGLES, IMUData);
+        rollValueInt = ((IMUData.data[0]<<8) & 0xFF00) + (IMUData.data[1] & 0xFF);
```

Comment:
i'd suggest pulling all the two-bytes-to-int code into a single function since it's always the same.  give it an offset and return an int.